src/main/java/executor/service/service/impl/ProxySourcesClientImpl.java

```diff
@@ -1,11 +1,59 @@
 package executor.service.service.impl;
 
+import executor.service.handler.ProxyHandler;
 import executor.service.model.ProxyConfigHolder;
+import executor.service.service.ProxyProvider;
 import executor.service.service.ProxySourcesClient;
+import executor.service.service.ProxyValidator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
 
 public class ProxySourcesClientImpl implements ProxySourcesClient {
+
+    private static final Logger log = LoggerFactory.getLogger(ProxySourcesClient.class);
+    private static final long DELAY = 1000;
+    private final ProxyProvider provider;
+    private final ProxyValidator validator;
+
+
+    public ProxySourcesClientImpl(ProxyProvider provider, ProxyValidator validator) {
+        this.provider = provider;
+        this.validator = validator;
+    }
+
+
     @Override
-    public ProxyConfigHolder getProxy() {
-        return null;
+    public void execute(ProxyHandler handler) {
+        List<ProxyConfigHolder> proxies = provider.readProxy();
+        validateList(proxies);
+        listen(proxies, handler);
+    }
+
+    private void validateList(List<ProxyConfigHolder> proxies) {
+        if(proxies == null || proxies.isEmpty()) {
+            log.error("Bad proxy list.");
+            throw new IllegalArgumentException("List cannot be null or empty");
+        }
+    }
+    private void listen(List<ProxyConfigHolder> proxies, ProxyHandler handler) {
+        int currentIndex = 0;
+        while(true){
```

Comment:
while' statement cannot complete without throwing an exception. This is not very good practice. 

In order to solve this problem, you need to redo the code following the example of ProxySourcesClient, pull-request [#43](https://github.com/GFLCourses5/worker/pull/43). 
This will allow in ParalleFlowExecutorService, pull-request [#30](https://github.com/GFLCourses5/worker/pull/30) to solve this problem correctly. 
Here, when calling the **public void shutdown()** method, the flag is switched correctly and the ExecutorService is completed.
![image](https://github.com/GFLCourses5/worker/assets/140727960/4a25fdc7-6b27-4f0f-bfb5-9b427e23b0d8)

Reply:
To my mind we dont need shutdown et all despite it is a good practice.

Reply:
Flux is about something else. ExecutorService requires **void shutdown()**.
You can see this if you read the Java documentation on ExecutorService